prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -477,6 +507,51 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
         }
     }
 
+    /**
+     * Encrypt the report to the REST service. This is a suspend function, meaning it can get called as an
+     * async method, though we call it a blocking way.
+     *
+     * @param message The report we want to send as a ByteArray
+     * @param encryptionKeyUrl The URL to extract the encryption key
+     * @param context Really just here to get logging injected
+     */
+    suspend fun encryptTheReport(
+        message: ByteArray,
+        fileName: String,
+        encryptionKeyUrl: String,
+        headers: Map<String, String>,
+        logger: Logger,
+        httpClient: HttpClient,
+    ): ByteArray {
+        logger.info(fileName)
+        httpClient.use { client ->
+            val encryptionKey = client.get(encryptionKeyUrl) {
+                postHeaders(
+                    // Calculate Content-Length if needed.
+                    headers.map { (key, value) ->
+                        if (key == "Content-Length" && value == "<calculated when request is sent>") {
```

Comment:
Understood, but when would "\<calculated when request is sent\>" be set as the header value? My understanding that's a Postman-only placeholder. Are we using that somewhere in the ReportStream application?

Reply:
Yes, it will be set in the REST transport message header.  In the latest RS's REST Transport, I introduced the authentication and message header in the transport.  Now, you can set the header key-value in the transport as Postman.

Reply:
I see now. I hadn't looked at the configs.